python/pyspark/rdd.py

```diff
@@ -2029,7 +2030,11 @@ def coalesce(self, numPartitions, shuffle=False):
         >>> sc.parallelize([1, 2, 3, 4, 5], 3).coalesce(1).glom().collect()
         [[1, 2, 3, 4, 5]]
         """
-        jrdd = self._jrdd.coalesce(numPartitions, shuffle)
+        if shuffle:
```

Comment:
Yeah to be consistent with the Scala side, it would be nice to rearrange this to have Python's `repartition(...)` call Python's `coalesce(..., shuffle=True)`.